rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -297,90 +455,88 @@ public static FlowNode getFlowEndNode(FlowExecution execution) {
+            cache.put(executionUrl, nodes);
+        }
+        return nodes;
     }
 
-    public static List<FlowNode> getUnsortedSortedExecutionNodeList(FlowExecution execution) {
-        return getUnsortedSortedExecutionNodeList(execution.getCurrentHeads());
-    }
 
-    static List<FlowNode> getUnsortedSortedExecutionNodeList(List<FlowNode> nodeList) {
-        if (nodeList == null || nodeList.isEmpty()) {
-            return Collections.EMPTY_LIST;
+    /** This is used to cover an obscure case where a WorkflowJob is renamed BUT
+     *  a previous WorkflowJob existed with cached execution data.
+     *  Otherwise the previous job's cached data would be returned.
+     **/
+    @Extension
+    public static class RenameHandler extends ItemListener {
+        @Override
+        public void onLocationChanged(Item item, String oldFullName, String newFullName) {
+            if (item instanceof WorkflowJob) {
+                // It's a cache, we can rebuild it... we have the technology.
+                CacheExtension ext = CacheExtension.all().get(0);
+                ext.getExecutionCache().invalidateAll();
+                ext.getRunCache().invalidateAll();
+            }
         }
+    }
 
-        FlowNodeListCacheAction cacheAction = FlowNodeListCacheAction.get(nodeList.get(0));
-        if (cacheAction.unsortedNodeList != null) {
-            return cacheAction.unsortedNodeList;
-        }
+    public static final Comparator<FlowNode> sortComparator = new Comparator<FlowNode>() {
+        @Override
+        public int compare(FlowNode node1, FlowNode node2) {
+            int node1Iota = parseIota(node1);
+            int node2Iota = parseIota(node2);
 
-        // Gather all the nodes from the workflow
-        final List<FlowNode> unsortedNodes = new ArrayList<FlowNode>();
-        FlowGraphWalker walker = new FlowGraphWalker(nodeList.get(0).getExecution());
-        for (FlowNode node : walker) {
-            if (!unsortedNodes.contains(node)) {
-                unsortedNodes.add(node);
+            if (node1Iota < node2Iota) {
+                return -1;
+            } else if (node1Iota > node2Iota) {
+                return 1;
             }
+            return 0;
         }
 
-        cacheAction.unsortedNodeList = unsortedNodes;
-
-        return cacheAction.unsortedNodeList;
-    }
-
-    private static List<FlowNode> sortNodesById(List<FlowNode> nodes) {
-        Comparator<FlowNode> sortComparator = new Comparator<FlowNode>() {
-            @Override
-            public int compare(FlowNode node1, FlowNode node2) {
-                int node1Iota = parseIota(node1);
-                int node2Iota = parseIota(node2);
-
-                if (node1Iota < node2Iota) {
-                    return -1;
-                } else if (node1Iota > node2Iota) {
-                    return 1;
-                }
+        private int parseIota(FlowNode node) {
+            try {
+                return Integer.parseInt(node.getId());
```

Comment:
Sure does!  You'll be happy to know this is being addressed in the new FlowAnalyzer.